dask_expr/_collection.py

```diff
@@ -328,6 +328,14 @@ def visualize(self, tasks: bool = False, **kwargs):
     def index(self):
         return new_collection(self.expr.index)
 
+    @index.setter
+    def index(self, value):
+        assert expr.are_co_aligned(
```

Comment:
Out of scope: Is the docstring of `are_co_aligned` inverted? From my superficial understanding, it should rather be `Do inputs come from the same parents, ignoring blockwise?` to match the `True`/`False` results.

Reply:
Yep :) Tells you how much I read my own docstrings (at least the function name is helpful :)). I'll adjust in my next pr